### DISCUSSION TITLE
HostGator SMTP Error: Could not connect to SMTP host

### DISCUSSION THREAD

**Discussion by kylerballiet** [PHPMailer/PHPMailer#3034] [Q&A] [answered]
I am getting the error `Could not connect to SMTP host` when trying to use PHPMailer package on HostGator.

I am attempting to upgrade to the new PHPMailer package since I am currently using the old package with PHPMailerAutoload but my web server is upgrading PHP to v8.3 which does not work with the old PHPMailer package.

I went through the [troubleshooting guide](https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting) but still cannot figure out what is going wrong. I verified I do have the `openssl` extension loaded so that is not the issue.

For testing, I am running this on the web server itself with PHP v8.3. Here is my code: 
```
date_default_timezone_set('Etc/UTC');
echo "Default timezone set to Etc/UTC\n";
echo (extension_loaded('openssl')?'SSL loaded':'SSL not loaded')."\n";

require 'PHPMailer-master/src/PHPMailer.php';
require 'PHPMailer-master/src/SMTP.php';
require 'PHPMailer-master/src/Exception.php';

$mail = new PHPMailer(true);

$mail->isSMTP();
$mail->SMTPDebug = SMTP::DEBUG_CONNECTION;
$mail->Debugoutput='html';
$mail->Host='myhost.hostgator.com';
$mail->Port=465;
$mail->SMTPAuth=true;
$mail->AuthType = 'LOGIN';
$mail->SMTPSecure='PHPMailer::ENCRYPTION_SMTPS';
$mail->Username='admin@mydomain.com';
$mail->Password='my_password';

$mail->setFrom('admin@mydomain.com');
$mail->addAddress('myemail@gmail.com');

$mail->isHTML(false);
$mail->Subject="My Subject";
$mail->Body="My Body";
```
The above returns this error log in the browser:
```
Default timezone set to Etc/UTCSSL loaded 2024-03-13 04:26:45 Connection: opening to myhost.hostgator.com:465, timeout=300, options=array()
2024-03-13 04:26:45 Connection: opened
2024-03-13 04:31:45 SERVER -> CLIENT:
2024-03-13 04:31:45 Connection: closing due to error
2024-03-13 04:31:45 Connection: closed
SMTP Error: Could not connect to SMTP host.
```
I also checked the `error_log` file for my website which had this:
```
[13-Mar-2024 04:31:45 Etc/UTC] PHP Fatal error:  Uncaught PHPMailer\PHPMailer\Exception: SMTP Error: Could not connect to SMTP host. in /home3/user/public_html/PHPMailer-master/src/PHPMailer.php:2286
Stack trace:
#0 /home3/user/public_html/PHPMailer-master/src/PHPMailer.php(2062): PHPMailer\PHPMailer\PHPMailer->smtpConnect(Array)
#1 /home3/user/public_html/PHPMailer-master/src/PHPMailer.php(1689): PHPMailer\PHPMailer\PHPMailer->smtpSend('Date: Wed, 13 M...', 'We have receive...')
#2 /home3/user/public_html/PHPMailer-master/src/PHPMailer.php(1523): PHPMailer\PHPMailer\PHPMailer->postSend()
#3 /home3/user/public_html/refillrequest_test.php(39): PHPMailer\PHPMailer\PHPMailer->send()
#4 {main}
  thrown in /home3/user/public_html/PHPMailer-master/src/PHPMailer.php on line 2286
```

I checked HostGator's [email connection settings](https://www.hostgator.com/help/article/email-connection-settings) to verify I am using the correct settings for SMTP - they use port 465.

This is the current production version that is working with PHP 7.4:
```
require 'phpmailer/PHPMailerAutoload.php';
$mail = new PHPMailer;

$mail->Host='myhost.hostgator.com';
$mail->Port=465;
$mail->SMTPAuth=true;
$mail->SMTPSecure='ssl';
$mail->Username='admin@mydomain.com';
$mail->Password='mypassword';

$mail->setFrom('admin@mydomain.com');
$mail->addAddress('myemail@gmail.com');

$mail->isHTML(false);
$mail->Subject="My subject";
$mail->Body="My body";
```
Any help would be greatly appreciated. Thank you.

**Comment by Synchro** [comment]
This looks like a TLS problem, which may be configured differently in your new PHP build – try the openssl tests outlined in the troubleshooting guide.

**Reply by kylerballiet** [reply, marked as answer]
Thanks for your prompt response.

I tried `ENCRYPTION_STARTTLS` on port `587` even though HostGator doesn't officially support it to see if it would give me a better error message because, like it says in the troubleshooting guide, the implicit encryption hides error messages making it hard to debug.

It ended up working and it spit out an error message about an authentication error due to invalid credentials. I was not expecting this because I am using those same credentials in my old script using a deprecated version of PHPMailer on PHP v7.4. I tried to log into the email server directly with the password I was using in the script and it didn't work so I reset the password and was able to log in. I updated the password in my new script using the latest version of PHPMailer on PHP v8.3 and now everything's working fine.

Glad it's working now but I'm curious as to why the old script continues to work with a password that's invalid. My guess would be the old configuration isn't requiring authentication since it's being run on the web server, which is a known host, but now the new configuration requires authentication regardless of where the request is coming from.
